dependencies/dependencies.go

```diff
@@ -50,12 +52,30 @@ func (d Dependencies) URLValidator() (url.Validator, error) {
 	return nil, errors.New(codes.Unimplemented, "url validator uninitialized in dependencies")
 }
 
+// newDefaultTransport creates a new transport with sane defaults.
+func newDefaultTransport() *http.Transport {
+	// These defaults are copied from http.DefaultTransport.
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			// DualStack is deprecated
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		MaxIdleConnsPerHost:   100,
```

Comment:
Looks good, but can you add a comment to this saying is not part of the defaults so we know which ones we have changed and which are from Go's defaults?